README.md

```diff
@@ -62,6 +62,43 @@ func main() {
 }
 ```
 
+## Migrating from 1.5.x to 2.x
+
+To upgrade to `ces-go-parser` version `2.x`, you need to update the parser version in your `go.mod` file and adjust all
+import paths in your project. 
+
+Specifically, append `/v2` to the import path, for example, update:
+```go
+    "github.com/make-software/ces-go-parser"
+```
+
+to:
+
+```go
+    "github.com/make-software/ces-go-parser/v2"
+```
+
+In version `2.x`, `ces-go-parser` supports event parsing from smart contracts defined as `AddressableEntity` on the network, leveraging the [GO SDK V2](https://github.com/make-software/casper-go-sdk/releases/tag/v2.0.2-beta1).
+By default, the parser will attempt to load smart contracts as `AddressableEntity`.
+
+If you need to use the new parser version with a `1.5.x` network, you can utilize the `NewParserWithVersion` constructor
+to specify the version, preventing an extra network calls to read the smart contract as an `AddressableEntity`.
+
+```go
+	parser, err := ces.NewParserWithVersion(rpcClient, []casper.Hash{contractHash}, ces.NetworkVersionV1)
+	if err != nil {
+		panic(err)
+	}
+```
+
+### Breaking changes
+
+The public interface of the `ces-go-parser` remains almost the same.
+
+There is only one interface change:
+
+- `func LoadContractMetadataWithoutSchema` accept list of `NamedKeys` and ContractPackageHash instead of `casper.Contract`
+
```

Comment:
Let's make it more user-oriented. 

## Migration to Casper 2.0.0 (Condor)

Casper 2.0.0 introduces changes in the API that aren't backward compatible with Casper 1.x. To use CES Go Parser with Casper 2.0.0 you need to the the version `v2` of the parser:

 ```go
     "github.com/make-software/ces-go-parser/v2"
 ```

If you want to use the same version of the parser for both Casper 1.x and Casper 2.x, you'll need to use the soft-migration constructor that requires you to specify the network version you are currently running on:

```go
 	parser, err := ces.NewParserWithVersion(rpcClient, []casper.Hash{contractHash}, ces.CasperVersion1x)
 	if err != nil {
 		panic(err)
 	}
 ```

### Breaking changes

In CES Go Parser 2.0.0, the `LoadContractMetadataWithoutSchema` function accepts a list of `NamedKeys` and ContractPackageHash instead of `casper.Contract` it accepted previously:

```
// @todo Example here
```